commands.go

```diff
@@ -722,16 +722,41 @@ func cmdEnv(c *cli.Context) {
 		}
 	}
 
+	usageHint := generateUsageHint(c, userShell)
+
 	switch userShell {
 	case "fish":
-		fmt.Printf("set -x DOCKER_TLS_VERIFY 1;\nset -x DOCKER_CERT_PATH %q;\nset -x DOCKER_HOST %s;\n",
-			cfg.machineDir, dockerHost)
+		fmt.Printf("set -x DOCKER_TLS_VERIFY 1;\nset -x DOCKER_CERT_PATH %q;\nset -x DOCKER_HOST %s;\n\n%s\n",
+			cfg.machineDir, dockerHost, usageHint)
 	default:
-		fmt.Printf("export DOCKER_TLS_VERIFY=1\nexport DOCKER_CERT_PATH=%q\nexport DOCKER_HOST=%s\n",
-			cfg.machineDir, dockerHost)
+		fmt.Printf("export DOCKER_TLS_VERIFY=1\nexport DOCKER_CERT_PATH=%q\nexport DOCKER_HOST=%s\n\n%s\n",
+			cfg.machineDir, dockerHost, usageHint)
 	}
 }
 
+func generateUsageHint(c *cli.Context, userShell string) string {
+	msg := "# Run this command to configure your shell: "
+	cmd := ""
+	machine := c.Args().First()
+
+	switch userShell {
+	case "fish":
+		if machine != "" {
+			cmd = fmt.Sprintf("eval (docker-machine env %s)", machine)
+		} else {
+			cmd = "eval (docker-machine env)"
+		}
+	default:
+		if machine != "" {
+			cmd = fmt.Sprintf("eval $(docker-machine env %s)", machine)
+		} else {
+			cmd = "eval $(docker-machine env)"
+		}
+	}
+
+	return fmt.Sprintf("%s %s\n", msg, cmd)
```

Comment:
We could also just have the message as part of the `Sprintf` to avoid the extra var too.